src/sinks/splunk_hec/common/acknowledgements.rs

```diff
@@ -0,0 +1,294 @@
+use std::{collections::HashMap, num::NonZeroU8, sync::Arc, time::Duration};
+
+use hyper::Body;
+use serde::{Deserialize, Serialize};
+use tokio::sync::{mpsc::UnboundedReceiver, oneshot::Sender};
+use vector_core::event::EventStatus;
+
+use crate::http::HttpClient;
+
+use super::service::HttpRequestBuilder;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct HecClientAcknowledgementsConfig {
+    pub query_interval: NonZeroU8,
+    pub retry_limit: NonZeroU8,
+}
+
+impl Default for HecClientAcknowledgementsConfig {
+    fn default() -> Self {
+        Self {
+            query_interval: NonZeroU8::new(10).unwrap(),
+            retry_limit: NonZeroU8::new(30).unwrap(),
+        }
+    }
+}
+
+pub fn default_hec_client_acknowledgements_config() -> Option<HecClientAcknowledgementsConfig> {
+    Some(HecClientAcknowledgementsConfig::default())
+}
+
+#[derive(Deserialize, Serialize, Eq, PartialEq, Debug)]
+pub struct HecAckQueryRequestBody {
```

Comment:
Note, there is some overlap in necessary structures with the [source-side implementation](https://github.com/vectordotdev/vector/pull/10044/files#diff-d4466a02a0d126c29c9f5f53abce0f98456dbc58c423210d4b80500dbd67f87cR214). Will consolidate these once the source-side implementation is merged.

Reply:
On second thought, the duplication is minor, and it seems better/simpler not to couple the source and sink in that way. This leaves room to adjust the sink-side implementation if needed to support future Splunk HEC versions while keeping our source-side implementation fixed. I'll leave it as is for now.